### PR TITLE
Add predicates for subsets, supersets, and disjoint sets

### PR DESCRIPTION
**Pull request by thass0**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -517,6 +517,91 @@ impl<T: PrimInt + Unsigned> SparseSet<T> {
...
+        true
+    }
+
+    /// Return `true` if `self` (A) is a proper subset of `other` (B):
+    ///
+    /// A ⊊ B.
+    ///
+    /// This is the same as [`Self::is_proper_superset`] with the arguments flipped.
+    pub fn is_proper_subset(&self, other: &Self) -> bool {
+        self.is_subset(other) && self.inner.len() < other.inner.len()
```
Comment (Chriscbr) [inline src/lib.rs]: Would it be faster to perform the size check first due to boolean short circuiting?

(in the case the result is `true`, both sides have to be computed so the order doesn't matter -- but in the case it's `false`, you'd rather perform the cheaper computation first)
```suggestion
        self.inner.len() < other.inner.len() && self.is_subset(other)
```

File: src/lib.rs [hunk 2]
```diff
@@ -517,6 +517,91 @@ impl<T: PrimInt + Unsigned> SparseSet<T> {
...
+    }
+
+    /// Return `true` if `self` (A) is a proper superset of `other` (B):
+    ///
+    /// A ⊋ B.
+    ///
+    /// This is the same as [`Self::is_proper_subset`] with the arguments flipped.
+    pub fn is_proper_superset(&self, other: &Self) -> bool {
+        self.is_superset(other) && self.inner.len() > other.inner.len()
```
Comment (Chriscbr) [inline src/lib.rs]: See comment above

File: src/lib.rs [hunk 3]
```diff
@@ -517,6 +517,91 @@ impl<T: PrimInt + Unsigned> SparseSet<T> {
...
+            }
+        }
+
+        for x in other.iter() {
+            if self.contains(x) {
+                return false;
+            }
+        }
+
+        true
```
Comment (Chriscbr) [inline src/lib.rs]: I think only the first loop is needed, unless I'm mistaken
```suggestion
        true
```

**Review by Chriscbr — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (thass0) [reply src/lib.rs]: Yes, it would! I'll update both cases

File: src/lib.rs (hunk 3 above)
Comment (thass0) [reply src/lib.rs]: Yes, you are correct, I think. If  the sets *A* and *B* are not disjoint, then they have some elements in common. Any element that they have in common can be found in *A* **and** in *B*, so, to run the `is_disjoint` check, it's sufficient to iterate over the elements of one set only. Regardless of which one, the common elements would be picked up.

**Pull request merged**